stratum/hal/lib/barefoot/bf_switch.cc

```diff
@@ -111,14 +111,16 @@ namespace {
 ::util::Status ConvertToLegacyForwardingPipelineConfig(
     const ::p4::v1::ForwardingPipelineConfig& forwarding_config,
     ::p4::v1::ForwardingPipelineConfig* legacy_config) {
-  *legacy_config = forwarding_config;
   BfPipelineConfig bf_config;
-  if (ExtractBfPipelineConfig(forwarding_config, &bf_config).ok()) {
+  if (bf_config.ParseFromString(forwarding_config.p4_device_config())) {
     std::string pi_p4_device_config;
     RETURN_IF_ERROR(
         BfPipelineConfigToPiConfig(bf_config, &pi_p4_device_config));
     legacy_config->set_p4_device_config(pi_p4_device_config);
+  } else {
+    *legacy_config = forwarding_config;
   }
+
```

Comment:
To me having the other code path in an explicit branch made the code easier to read, and it matches the comment (Otherwise, ...)